src/NLog/Internal/XmlHelper.cs

```diff
@@ -288,57 +308,49 @@ internal static string XmlConvertToString(object value, TypeCode objTypeCode, bo
                 case TypeCode.UInt64:
-                    return XmlConvert.ToString((ulong)value);
+                    return XmlConvert.ToString(value.ToUInt64(CultureInfo.InvariantCulture));
                 case TypeCode.Single:
                     {
-                        float singleValue = (float)value;
+                        float singleValue = value.ToSingle(CultureInfo.InvariantCulture);
                         if (float.IsInfinity(singleValue))
-                            return Convert.ToString(singleValue, System.Globalization.CultureInfo.InvariantCulture); // Infinity instead of INF
+                            return Convert.ToString(singleValue, CultureInfo.InvariantCulture); // Infinity instead of INF
                         else
                             return XmlConvert.ToString(singleValue);    // 8 digits scale
                     }
                 case TypeCode.Double:
                     {
-                        double doubleValue = (double)value;
+                        double doubleValue = value.ToDouble(CultureInfo.InvariantCulture);
                         if (double.IsInfinity(doubleValue))
-                            return Convert.ToString(doubleValue, System.Globalization.CultureInfo.InvariantCulture); // Infinity instead of INF
+                            return Convert.ToString(doubleValue, CultureInfo.InvariantCulture); // Infinity instead of INF
                         else
                             return XmlConvert.ToString(doubleValue);    // 16 digits scale
                     }
                 case TypeCode.Decimal:
-                    return XmlConvert.ToString((decimal)value);
+                    return XmlConvert.ToString(value.ToDecimal(CultureInfo.InvariantCulture));
                 case TypeCode.DateTime:
-                    return XmlConvert.ToString((DateTime)value, XmlDateTimeSerializationMode.Utc);
+                    return XmlConvert.ToString(value.ToDateTime(CultureInfo.InvariantCulture), XmlDateTimeSerializationMode.Utc);
                 case TypeCode.Char:
-                    return XmlConvert.ToString((char)value);
+                    return XmlConvert.ToString(value.ToChar(CultureInfo.InvariantCulture));
                 case TypeCode.String:
-                    return safeConversion ? RemoveInvalidXmlChars((string)value) : (string)value;
+                    return safeConversion ? RemoveInvalidXmlChars(value.ToString(CultureInfo.InvariantCulture)) : value.ToString(CultureInfo.InvariantCulture);
                 default:
-                    try
-                    {
-                        string valueString = Convert.ToString(value, System.Globalization.CultureInfo.InvariantCulture);
-                        return safeConversion ? RemoveInvalidXmlChars(valueString) : valueString;
-                    }
-                    catch
-                    {
-                        return null;
-                    }
+                    return null;
```

Comment:
please create a variable for CultureInfo.InvariantCulture (too avoid clutter and long lines)

Reply:
Think it makes the code less readable to take global static variables and storing them in local-variables (when there is no fallback logic)